File: сross_platform_software_systems/14/user_service/user_service/src/main/java/com/c3equalz/user_service/application/common/ports/AccessRevoker.java
package com.c3equalz.user_service.application.common.ports;

import com.c3equalz.user_service.domain.user.values.UserID;
import reactor.core.publisher.Mono;

/**
 * Interface for revoking user access.
 */
public interface AccessRevoker {

    /**
     * Removes all access for the specified user.
     * This is typically used when a user cannot be found or is invalid.
     *
     * @param userId the user ID whose access should be revoked
     * @return Mono that completes when access revocation is done
     */
    Mono<Void> removeAllUserAccess(UserID userId);
}

File: сross_platform_software_systems/14/chat_service/src/chat_service/application/common/ports/access_revoker.py
from abc import abstractmethod
from typing import Protocol

from chat_service.domain.user.values.user_id import UserID


class AccessRevoker(Protocol):
    @abstractmethod
    async def remove_all_user_access(self, user_id: UserID) -> None:
        """
        :raises DataMapperError:
        """
